src/syft/core/plan/translation/torchscript/plan_translate.py

```diff
@@ -0,0 +1,108 @@
+# stdlib
+from collections import OrderedDict
+from typing import Any
+from typing import List
+
+# third party
+import torch as th
+
+# syft absolute
+from syft.core.plan import Plan
+from syft.core.plan.plan_builder import PLAN_BUILDER_VM
+from syft.core.plan.plan_builder import ROOT_CLIENT
+from syft.lib.python.collections import OrderedDict as SyOrderedDict
+from syft.lib.python.dict import Dict
+from syft.lib.python.list import List as SyList
+from syft.logger import traceback_and_raise
+
+# syft relative
+from ....pointer.pointer import Pointer
+from .plan import PlanTorchscript
+
+
+def translate(plan: Plan) -> PlanTorchscript:
+    """Translates Syft Plan to torchscript"""
+
+    class PlanWrapper(th.nn.Module):
+        """
+        Plan needs to be executed with kwargs while torchscript needs to be executed with args,
+        so we need to have kwarg names inside torchscript to pass them into Plan.
+        We use nn.Module to store input kwarg names inside torchscript.
+        """
+
+        def __init__(self, kwarg_names: List[str]):
+            _kwarg_names: List[str]
+            super(PlanWrapper, self).__init__()
+            self._kwarg_names = kwarg_names
+
+        def forward(self, *args: Any) -> Any:
+            kwarg_ptrs: Dict[str, Pointer] = {}
+
+            # Since Syft Plan needs pointers as args,
+            # reverse-map actual arg values to pointers
+            for name, arg in zip(self._kwarg_names, args):
+                found = False
+                for ptr in ROOT_CLIENT.store:
+                    store_value: Any = PLAN_BUILDER_VM.store[ptr.id_at_location].data
+                    is_list = isinstance(arg, (SyList, list)) and isinstance(
+                        store_value, (SyList, list)
+                    )
+                    is_dict = isinstance(
+                        arg, (dict, Dict, OrderedDict, SyOrderedDict)
```

Comment:
could we define some constants somewhere? Like: LIST_TYPES = (SyList, list), DICT_TYPES = (dict, Dict, OrderedDict, SyOrderedDict)? you can store these in syft/lib/python?

Reply:
Yeah it would be nice to have Unions of these be auto generated for these kinds of checks.

Reply:
`OrderedDict` is a `dict` so you don't really need it in the list:
```python
from collections import OrderedDict
isinstance(OrderedDict(), dict)
Out[8]: True
```

`SyOrderedDict` is an `OrderedDict` so it's not really needed either unless you think that might changed in the future but I assume it will always at least be a `dict`.

Looking around a bit online, it looks like you don't really need `Dict` in there either: https://stackoverflow.com/questions/62547848/should-isinstance-check-against-typing-or-collections-abc

so `isinstance(arg, dict)` should be sufficient.